docs/source/developers.md

```diff
@@ -1,4 +1,4 @@
-# Notes for developers
+# Setting up the Development Environment
 
 ## Development environment
```

Comment:
I think it's not clear how to actually run the tests. I am currently trying to do that and I wonder "Can I run the tests without an Azure account?" "Can I just run `pytest` from the root?".

Reply:
Maybe a quickstart to the commands in the different `Makefile`s would help.

Reply:
In this PR, I only wanted to get the coding guidelines in. Improving Dev onboarding is a different question.